From: Athos Ribeiro <athos.ribeiro@canonical.com>
Date: Sat, 20 Jan 2024 14:01:24 +0100
Subject: Disable GC in ini files

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=831752
Bug-Ubuntu: https://bugs.launchpad.net/debian/+source/php7.3/+bug/1772915
Forwarded: not-needed
Last-Update: 2024-01-19

This is disabled in the Debian packages due to the strict permissions on
/var/lib/php. Instead, GC is performed through /etc/cron.d/php, which uses the
session.gc_maxlifetime setting.
Last-Update: 2024-01-19
---
 php.ini-development | 7 ++++++-
 php.ini-production  | 7 ++++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/php.ini-development b/php.ini-development
index 9caf434..6602495 100644
--- a/php.ini-development
+++ b/php.ini-development
@@ -1377,7 +1377,12 @@ session.serialize_handler = php
 ; Development Value: 1
 ; Production Value: 1
 ; https://php.net/session.gc-probability
-session.gc_probability = 1
+; Debian Default Value: 0
+; This is disabled in the Debian packages due to the strict permissions
+; on /var/lib/php. Instead, GC is performed through /etc/cron.d/php,
+; which uses the session.gc_maxlifetime setting. Please, check
+; /usr/share/doc/php8.3-common/README.Debian.gz for further reference.
+session.gc_probability = 0
 
 ; Defines the probability that the 'garbage collection' process is started on every
 ; session initialization. The probability is calculated by using gc_probability/gc_divisor,
diff --git a/php.ini-production b/php.ini-production
index bf3e914..a53265a 100644
--- a/php.ini-production
+++ b/php.ini-production
@@ -1373,7 +1373,12 @@ session.serialize_handler = php
 ; Development Value: 1
 ; Production Value: 1
 ; https://php.net/session.gc-probability
-session.gc_probability = 1
+; Debian Default Value: 0
+; This is disabled in the Debian packages due to the strict permissions
+; on /var/lib/php. Instead, GC is performed through /etc/cron.d/php,
+; which uses the session.gc_maxlifetime setting. Please, check
+; /usr/share/doc/php8.2-common/README.Debian.gz for further reference.
+session.gc_probability = 0
 
 ; Defines the probability that the 'garbage collection' process is started on every
 ; session initialization. The probability is calculated by using gc_probability/gc_divisor,
